File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(int r1, int r2) {
    long long answer = 0;
    
    long long r1_pow = (long long)pow(r1, 2);
    long long r2_pow = (long long)pow(r2, 2);
    for(int i = 1; i <= r2; i++) {
        long long i_pow = (long long)pow(i, 2);
        
        int minY = ceil(sqrt(r1_pow - i_pow));
        int maxY = floor(sqrt(r2_pow - i_pow));

        if (r1 < i)
            minY = 0;

        answer += maxY - minY + 1;
    }
    
    answer *= 4;
    
    return answer;
}

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.js
function solution(r1, r2) {
    let answer = 0;
    
    const r1_pow = Math.pow(r1, 2);
    const r2_pow = Math.pow(r2, 2);
    
    for(let i = 1; i <= r2; i++) {
        let i_pow = Math.pow(i, 2);
        
        let minY = Math.ceil(Math.sqrt(r1_pow - i_pow));
        let maxY = Math.floor(Math.sqrt(r2_pow - i_pow));
        
        if(r1 < i)
            minY = 0;
        
        answer += maxY - minY + 1;
    }
    
    answer *= 4;
    
    return answer;
}